manifest/manifestlist/manifestlist_test.go

```diff
@@ -109,3 +110,105 @@ func TestManifestList(t *testing.T) {
 		}
 	}
 }
+
+// TODO (mikebrow): add annotations on the index and individual manifests
+var expectedOCIImageIndexSerialization = []byte(`{
```

Comment:
Oops should have refreshed before sending...

@owtaylor, I don't have a particular use-case in mind, but I think it makes sense to be deliberate about supporting or not supporting certain features. It seems arbitrary not to support distributing an image index of this form, since it's not actually incompatible with the distribution model (just the implementation). If we agree that this is a **bug**, I'm happy :)

Given that registry tags (currently) exist outside of the image's merkle dag, I don't see any reason to forbid tagging an Image Index and pulling it in the same way you would tag and pull a Manifest List. The registry would just round-trip the image index. If we want to explicitly handle "org.opencontainers.image.ref.name" annotations as a registry concept, I'd love to see some kind of (draft?) spec describing the expected behavior. In its current form, I'd expect a some other system (or new registry API v3?) to consume those refs.

Alternatively, we could _relax_ the handling of Manifest Lists such that platform is not a required attribute. This would make them mostly compatible with Image Indexes (I believe). As I understand it, the docker specs are more descriptive than prescriptive, but I'll defer to @stevvooe